tlsconfig/config.go

```diff
@@ -83,6 +84,22 @@ func certPool(caFile string) (*x509.CertPool, error) {
 	return certPool, nil
 }
 
+// minVersion returns the minimum TLS version matching 'version', the MinVersion string.
+func minVersion(version string) (uint16, error) {
+	allVersions := map[string]uint16{
+		"VersionSSL30": tls.VersionSSL30,
```

Comment:
Agreed. @adshmh would you mind removing this function from here and pass in a `uint16` instead of string?